pkg/controller/servicebinding/binder.go

```diff
@@ -436,11 +450,20 @@ func (b *binder) removeContainer(container interface{}) (map[string]interface{},
 // appendVolumeMounts append the binding volume in the template level.
 func (b *binder) appendVolumeMounts(volumeMounts []corev1.VolumeMount) []corev1.VolumeMount {
 	name := b.sbr.GetName()
-	mountPath := b.sbr.Spec.MountPathPrefix
+	mountPath := b.bindingRootEnv
+	fixedMountPath := false
 	if mountPath == "" {
-		mountPath = "/var/data"
+		mountPath = b.sbr.Spec.MountPath
+		if mountPath == "" {
+			mountPath = "/bindings"
+		} else {
+			fixedMountPath = true
+		}
 	}
 
+	if !fixedMountPath {
+		mountPath = path.Join(mountPath, name)
+	}
```

Comment:
for better readability and maintainability, let's extract this logic into `MountPath` function defined on `ServiceBindingSpec` struct:

```golang
func (spec *ServiceBindingSpec) MountPath(appContainer *v1.Container)
```

and with that `appendVolumeMounts` becomes:

```golang
func (b *binder) appendVolumeMounts(volumeMounts []corev1.VolumeMount, mounthPath string) []corev1.VolumeMount {
for _, v := range volumeMounts {
		if name == v.Name {
			return volumeMounts
		}
	}

	return append(volumeMounts, corev1.VolumeMount{
		Name:      name,
		MountPath: mountPath,
	})
}
```

Reply:
Done. f77cceca5df84535986fd9cbbf09425d265dbd98